### ISSUE TITLE
gRPC-cpp and gRPC-Core Value conversion build warnings

### ISSUE THREAD

**Issue by artembolotov** [firebase/firebase-ios-sdk#9790]
After clean build there are 131 warnigns from this dependences

**Comment by rizafran** [comment]
Hi @artembolotov, please provide the required information from the issue template to investigate the issue. Thanks.

**Comment by Peanutsmasher** [comment]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.3.1
  * Firebase SDK version: 8.15.0
  * Installation method: `Swift Package Manager`
  * Firebase Component: AppCheck, Firestore, Messaging, Storage
  * Target platform(s): `iOS`

### [REQUIRED] Step 2: Describe the problem

#### Steps to reproduce:

After updating to the latest Firebase SDK version, Xcode shows 392 gRPC-Core issues when compiling.
gRPC is on version 1.44.0-grpc
No prior issues, only after appeared after the update. Cleaning the build folder did not help.

Xcode states:
Value conversion issue - Implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long long') to 'uint32_t' (aka 'unsigned int')

#### Relevant Code:

``` UPB_INLINE uint64_t _upb_be_swap64(uint64_t val) {
  if (_upb_isle()) {
    return val;
  } else {
    return ((uint64_t)_upb_be_swap32(val) << 32) | _upb_be_swap32(val >> 32);
  }
}```

**Comment by artembolotov** [comment]
Thank you!

**Comment by rizafran** [comment]
Thanks for the info. I was able to reproduce the issue using the latest version. I'll inform the engineer about this, and we'll see what we can do.

**Comment by artembolotov** [comment]
I have this issue in latest version

**Comment by sidkamaria1337** [comment]
Having the same issue. I think it happens only after adding `FirebaseFirestore` dependency.

**Comment by paulb777** [comment]
We've had discussions with the Firebase dependency -  gRPC team and they are working to fix it in their next release. We'll update here when that gRPC release is available.

**Comment by paulb777** [comment]
This should now be fixed with an update to 1.44.2-grpc

**Comment by kennedycraig** [comment]
Hi Paul,
I just update all packages including to 1.44.2-grpc. While the number of warnings have reduced for me from >300, there are still see 124 warnings generated.
Xcode 13.4, Ios 15.5.
Sample screenshot attached. What info can I provide to help?
<img width="610" alt="image" src="https://user-images.githubusercontent.com/10380669/169560482-ddde3c1d-f12f-430e-bf5a-750fe6a67b58.png">

**Comment by paulb777** [comment]
Thanks @kennedycraig. I'll reopen and we'll follow up with the gRPC team and let you know.  cc: @dennycd

**Comment by RiverboatGambler** [comment]
Forgive me if I'm "not doing it right" because I'm new to this type of posting. 

I just opted to switch to a Firebase implementation using XCode Package Manager. Now, when I build (for Swift), I have a slew of warning related to this library. Since I'm clueless about protocol, I figured I would just start with this screenshot from XCode. 

Are these libraries safe to use in a Production app? Or, should I revert back to the Frameworks upon which I was relying? What is the preferred way of reporting these issues?

I am using XCode 13.4

<img width="349" alt="image" src="https://user-images.githubusercontent.com/59775247/171520256-fa4ba3f4-98d9-489f-a02e-421f84a31e86.png">

**Comment by paulb777** [comment]
@RiverboatGambler The only difference between the different installation methods is whether or not build warnings are suppressed. As noted above, the fixes for the warnings are in progress.

**Comment by RiverboatGambler** [comment]
Thanks, Paul. I'll take that as "nothing to worry about". Appreciate you and the amazing job the folks out here have done. Firebase has been a treasure trove for me on my first IOS app. Kudos to all of you for the great work!

**Comment by Peanutsmasher** [comment]
@paulb777 may I kindly ask whether there is any update? Fully aware that it's only warnings, just love to see no issues when compiling haha

**Comment by paulb777** [comment]
Three of the Firestore dependencies were updated today to address warnings in SPM builds. Update to 0.20220203.2 for https://github.com/firebase/abseil-cpp-SwiftPM, 0.9.1 for https://github.com/firebase/boringSSL-SwiftPM, and 1.44.3-grpc tag for  https://github.com/grpc/grpc-ios/

Note that these repos still have analyzer issues and build warnings remain for leveldb.

**Comment by paulb777** [comment]
I'm going to close since the gRPC build warnings are now addressed. Please open another issue if you'd like to track the remaining gRPC/abseil/BoringSSL analyzer issues (in progress) or the leveldb build warnings(not currently planned).
